weld/subsystem/src/main/java/org/jboss/as/weld/deployment/processors/WeldDeploymentProcessor.java

```diff
@@ -293,18 +294,20 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
         ServiceBuilder<WeldStartCompletionService> weldStartCompletionServiceBuilder = serviceTarget
                 .addService(weldStartCompletionServiceName, weldStartCompletionService)
                 .addDependency(weldBootstrapServiceName, WeldBootstrapService.class, weldStartCompletionService.getBootstrap())
-                .addDependency(weldStartServiceName).addDependencies(getComponentStartServiceNames(deploymentUnit));
+                .addDependency(weldStartServiceName);
+        getComponentStartServiceNames(deploymentUnit).entrySet().forEach(e->weldStartCompletionServiceBuilder.addDependencies(e.getKey(), e.getValue()));
         for (DeploymentUnit sub : subDeployments) {
-            weldStartCompletionServiceBuilder.addDependencies(getComponentStartServiceNames(sub));
+            getComponentStartServiceNames(sub).entrySet().forEach(e->weldStartCompletionServiceBuilder.addDependencies(e.getKey(), e.getValue()));
```

Comment:
`ServiceBuilder.addDependencies()` is deprecated and even says that optional dependencies are unsafe and should not be used.

Reply:
Can you provide recipe how to depend on component when it can not start in any moment?

Reply:
I don't know if it is possible at all. With current setup, Weld can do some cleanup operation allowing us to lower the footprint. If we cannot rely on all components started, then we cannot do any cleanup. It also seems that `org.jboss.msc.service.ServiceTarget` doesn't support your desired approach at all.

Can you please provide a reproducer for your problem (via JIRA - https://issues.jboss.org/browse/WFLY-10784)? If we had that, we could start fiddling with it and see what we can do.

Reply:
I added commit with reproducer, but can't run any test in wildlfy source tree, only in separate project